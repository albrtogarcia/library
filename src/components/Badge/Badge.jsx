import PropTypes from 'prop-types'
import './Badge.scss'

export const Badge = ({ type, text }) => <span className={`badge badge-${type}`}>{text}</span>

Badge.propTypes = {
  type: PropTypes.oneOf(['success', 'warning', 'danger', 'info']),
  text: PropTypes.string.isRequired
}
