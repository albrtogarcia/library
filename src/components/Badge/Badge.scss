.badge {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 0.25em 1em;
  font-size: .85rem;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  border-radius: 0.25rem;
  color: inherit;
  background-color: transparent;
  border: 1px solid transparent;

  &.badge-danger {
    background-color: var(--brand-red, #E31C3D);
    border-color: var(--brand-red, #E31C3D);
    color: var(--white, #FFF);
  }

  &.badge-warning {
    background-color: var(--brand-yellow, #FFC20E);
    border-color: var(--brand-yellow, #FFC20E);
    color: var(--white, #FFF);
  }

  &.badge-info {
    background-color: var(--brand-blue, #0189D5);
    border-color: var(--brand-blue, #0189D5);
    color: var(--white, #FFF);
  }

  &.badge-success {
    background-color: var(--brand-green, #00A859);
    border-color: var(--brand-green, #00A859);
    color: var(--white, #FFF);
  }
}