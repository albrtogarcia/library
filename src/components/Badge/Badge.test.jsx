import { describe, expect, it } from 'vitest'
import { Badge } from './Badge.jsx'

describe('Badge', () => {
  const props = {
    type: 'success',
    text: 'Success'
  }

  it('matches snapshot', () => {
    const badge = Badge(props)
    expect(badge).toMatchSnapshot()
  })
})
