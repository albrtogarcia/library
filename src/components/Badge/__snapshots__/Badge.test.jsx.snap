// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > matches snapshot 1`] = `
<span
  className="badge badge-success"
>
  Success
</span>
`;
